### PR TITLE
Fix minting

### PR DESCRIPTION
**Pull request by DannyDelott**
Minting in devnet should use the amount argument, while minting in sepolia should mint 1/2 the max mint amount.

Follow up to #921

### CODE REVIEW

File: apps/hyperdrive-trading/src/network/wagmiClient.ts [hunk 1]
```diff
@@ -49,6 +46,9 @@ if (
 if (VITE_SEPOLIA_RPC_URL) {
   chains.push(sepolia);
   transports[sepolia.id] = http(VITE_SEPOLIA_RPC_URL);
+  if (capsuleWallet) {
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/network/wagmiClient.ts:49]: Pinging capsule team, this still only works on sepolia

File: apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx [hunk 2]
```diff
@@ -34,10 +42,21 @@ export function useMintToken({
...
-  const mintAmount =
-    maxMintAmount !== undefined ? dnum.divide(maxMintAmount, [2n, 0])[0] : 0n;
+  let mintAmount = amount;
+
+  // On sepolia, only let users mint half the max amount
+  if (chainId === sepolia.id) {
+    if (maxMintAmount) {
+      mintAmount = maxMintAmount / 2n;
+    } else {
+      mintAmount = 0n;
```
Comment (jackburrus) [inline apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx:57]: we still want mintAmount to be some amount on devnet.

**Review by jackburrus — APPROVED** [review]

File: apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx (hunk 2 above)
Comment (DannyDelott) [reply apps/hyperdrive-trading/src/ui/token/hooks/useMintToken.tsx:57]: Yep, by default it is the `amount` from arguments.

**Review by ryangoree — APPROVED** [review]
Looks good, but I'd like to find out why this happening and if it's unique to Sepolia.

**Pull request merged**